onvif/client.py

```diff
@@ -419,16 +428,31 @@ def get_definition(self, name, port_type=None):
 
     def create_onvif_service(self, name, port_type=None):
         """Create ONVIF service client"""
-
         name = name.lower()
-        xaddr, wsdl_file, binding_name = self.get_definition(name, port_type)
-
         # Don't re-create bindings if the xaddr remains the same.
         # The xaddr can change when a new PullPointSubscription is created.
-        binding_key = f"{binding_name}{xaddr}"
-        binding = self.services.get(binding_key)
-        if binding:
-            return binding
+        binding_key = (name, port_type)
+
+        xaddr, wsdl_file, binding_name = self.get_definition(name, port_type)
+
+        existing_service = self.services.get(binding_key)
+        if existing_service:
```

Comment:
I think this is safe since there shouldn't be different xaddrs for the same namespace?